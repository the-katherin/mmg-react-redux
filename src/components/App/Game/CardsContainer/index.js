import './CardsContainer.css';

export { default } from './CardsContainer';