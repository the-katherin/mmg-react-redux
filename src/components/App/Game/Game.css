.game-window {
  display: grid;
  justify-content: center;
  grid-row-gap: 10px; 
  padding: 30px;
}

.game-window__buttons {
  display: grid;
  justify-content: left;
  grid-auto-flow: column;
  grid-column-gap: 10px; 

}
