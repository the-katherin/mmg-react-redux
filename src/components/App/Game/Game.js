import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';
import CardsContainer from './CardsContainer'
import Timer from './Timer';
import { getUserSelection, getIsFinished } from "../../../selectors";
import { generateCards } from '../../../actions/generateCards';

export class Game extends Component {


    constructor(props) {
        super(props);

        this.state = {
            redirect: false
        }

        this.props.generateCards(this.props.userSelection.level);


    }



    componentDidUpdate() {
        if (this.props.isFinished) {

            this.setState({ redirect: true });
        }
    }



    render() {

        const { redirect } = this.state;

        if (redirect) {
            return <Redirect to='/results' />;
        }


        return (
            <div className="game-window">
                <Timer />
                <CardsContainer />
                <div className="game-window__buttons">
                    <Link to='/results'><button>Results</button></Link>
                    <Link to='/form'><button>Change selection</button></Link>
                </div>

            </div>
        )
    }
};

Game.propTypes = {
    userSelection: PropTypes.shape({
        level: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        cardsBack: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    }).isRequired,

    generateCards: PropTypes.func.isRequired

};

export default connect(
    state => ({
        userSelection: getUserSelection(state),
        isFinished: getIsFinished(state),
    }),

    {

        generateCards
    }

)(Game);

