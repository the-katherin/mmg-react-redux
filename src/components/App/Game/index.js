import './Game.css';

export { default } from './Game';
export { Game } from './Game';