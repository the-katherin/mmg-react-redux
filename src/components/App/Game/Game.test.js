import React from "react";
import { Game } from "./Game";

let wrapper;
const generateCards = jest.fn();

describe("Game", () => {
    beforeEach(() => {
        wrapper = shallow(<Game
            userSelection={{ level: 6, cardsBack: 2 }}
            isFinished={false}
            generateCards={generateCards} />)
    });

    test("match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });


});
