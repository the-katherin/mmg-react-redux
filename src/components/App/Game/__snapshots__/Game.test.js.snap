// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game match snapshot 1`] = `
<div
  className="game-window"
>
  <Connect(Timer) />
  <Connect(CardsContainer) />
  <div
    className="game-window__buttons"
  >
    <Link
      replace={false}
      to="/results"
    >
      <button>
        Results
      </button>
    </Link>
    <Link
      replace={false}
      to="/form"
    >
      <button>
        Change selection
      </button>
    </Link>
  </div>
</div>
`;
